### PR TITLE
Formatting with fourmolu

### PR DESCRIPTION
**Pull request by andreabedini**

### CODE REVIEW

File: .github/workflows/formatting.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  build:
+    runs-on:
+      - ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: haskell-actions/run-fourmolu@v9
+        with:
+          version: "0.14.0.0"
```
Comment (yvan-sraka) [inline .github/workflows/formatting.yaml:16]: That's nice to have the `fourmolu` version pinned here and to have the CI check the formatting :)

Comment (yvan-sraka) [reply .github/workflows/formatting.yaml:16]: Maybe we can also have [`nix fmt`](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-fmt) set up to format the codebase with `haskellPackages.fourmolu_0_14_0_0` :)

**Review by yvan-sraka — APPROVED** [review]
This will likely make a lot of merge-conflict as side effect and break a bit `git blame` history, but that's nice! Just we better merge the other opened PR before this one right?

File: .github/workflows/formatting.yaml (hunk 1 above)
Comment (michaelpj) [reply .github/workflows/formatting.yaml:16]: We have a nix dev shell, we should make sure this version of `fourmolu` is available there. And then we can just run that. c.f. https://github.com/haskell/lsp/pull/519

Comment (michaelpj) [reply .github/workflows/formatting.yaml:16]: But I can tweak this later

**Review by michaelpj — APPROVED** [review]
Merge conflicts are easy to fix, just reset to "ours" and reformat.

File: .github/workflows/formatting.yaml (hunk 1 above)
Comment (andreabedini) [reply .github/workflows/formatting.yaml]: Added fourmolu to the devshell.

**Review by michaelpj — APPROVED** [review]

**Pull request merged**